src/chi/javascript/components/tooltip.js

```diff
@@ -39,21 +40,37 @@ class Tooltip extends Component {
       this._animationTimeout = window.setTimeout(() => {
         if (!self._shown) {
           self.show();
+          window.tooltipOpen = true;
+          clearTimeout(window.tooltipOpenTimeout);
         }
-      }, ANIMATION_DELAY);
+      }, window.tooltipOpen ? 0 : ANIMATION_DELAY);
     });
     this._addEventHandler(this._elem, 'mouseleave', () => {
       window.clearTimeout(this._animationTimeout);
       this._animationTimeout = null;
       self._hovered = false;
       if (self._shown && !self._focused) {
         self.hide();
+        window.tooltipOpenTimeout = setTimeout(() => {
+          if (window.tooltipOpen) {
+            window.tooltipOpen = false;
+          }
+        }, TOOLTIP_SWITCH_TIMEOUT);
       }
     });
     this._addEventHandler(this._elem, 'focus', function() {
       self._focused = true;
-      if (!self._shown) {
-        self.show();
+      self._addEventHandler(self._elem, 'keyup', function(e) {
+        let code = (e.keyCode ? e.keyCode : e.which);
+
+        if (code === 9) {
```

Comment:
Which key is coded with 9? Maybe a small comment will be handy here

Reply:
it's `Tab` key.
Good point, what if we define a reusable constant, something like:

```
export const keys = {
  tab: 9,
  left: 37,
  up: 38,
  right: 39,
  down: 40,
  delete: 46
};
```

to be imported in keyboard controlled components?

Reply:
Yes, good idea! I'd define the names in caps, though. 👍🏻

Reply:
Done, thanks! 😉

Reply:
Much better! 👍🏻